pkg/consuldp/consul_dataplane.go

```diff
@@ -180,8 +180,15 @@ func (cdp *ConsulDataplane) Run(ctx context.Context) error {
 	cdp.dpServiceClient = pbdataplane.NewDataplaneServiceClient(state.GRPCConn)
 
 	doneCh := make(chan error)
+	bootstrapParams, err := cdp.getBootstrapParams(ctx)
```

Comment:
get bootstrap params separately so it can be passed into `startDNSProxy()` and `bootstrapConfig()` calls separately.  This gets rid of relying on dataplane's `resolvedProxyConfig` to have been set prior and also the code reader and linter know this function relies on bootstrap params having been retrieved.